src/main/java/org/folio/rest/impl/FinanceHelper.java

```diff
@@ -0,0 +1,210 @@
+    return (map, fund) -> map.merge(fund.getLedgerId(), groupedByFund.get(fund.getId()), (transactions, transactions2) -> {
+      transactions.addAll(transactions2);
+      return transactions;
+    });
+  }
+
+  private CompletableFuture<List<Fund>> getFundsByIds(List<String> ids) {
+    String query = convertIdsToCqlQuery(ids);
+    String queryParam = "&query=" + encodeQuery(query, logger);
+    String endpoint = String.format(GET_FUNDS_WITH_SEARCH_PARAMS, MAX_IDS_FOR_GET_RQ, 0, queryParam, lang);
+
+    return HelperUtils.handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger)
+      .thenApply(entries -> entries.mapTo(FundCollection.class))
+      .thenApply(fundCollection -> {
+        if (ids.size() == fundCollection.getFunds().size()) {
+          return fundCollection.getFunds();
+        }
+        String missingIds = String.join(", ", CollectionUtils.subtract(ids, fundCollection.getFunds().stream().map(Fund::getId).collect(toList())));
+        throw new HttpException(400, FUNDS_NOT_FOUND.toError().withParameters(Collections.singletonList(new Parameter().withKey("funds").withValue(missingIds))));
+      });
+  }
+
+  private List<Pair<Transaction, FundDistribution>>  buildEncumbrances(CompositePurchaseOrder compPo) {
+
+    return compPo.getCompositePoLines()
+      .stream()
+      .flatMap(poLine -> poLine.getFundDistribution()
+        .stream()
+        .map(fundDistribution -> new MutablePair<>(buildEncumbrance(fundDistribution, poLine, compPo), fundDistribution)))
+      .collect(toList());
+  }
+
+  private CompletableFuture<FiscalYear> getCurrentFiscalYear(String ledgerId) {
+    String endpoint = String.format(GET_CURRENT_FISCAL_YEAR_BY_ID, ledgerId, lang);
+    return handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger).thenApply(entry -> entry.mapTo(FiscalYear.class))
+      .exceptionally(t -> {
+        if (isFiscalYearNotFound(t)) {
+          List<Parameter> parameters = Collections.singletonList(new Parameter().withValue(ledgerId)
+            .withKey("ledgerId"));
+          throw new HttpException(400, CURRENT_FISCAL_YEAR_NOT_FOUND.toError()
+            .withParameters(parameters));
+        }
+        throw new CompletionException(t.getCause());
+      });
+  }
+
+  private boolean isFiscalYearNotFound(Throwable t) {
+    return t.getCause() instanceof HttpException && ((HttpException) t.getCause()).getCode() == 404;
+  }
+
+  private Transaction buildEncumbrance(FundDistribution distribution, CompositePoLine poLine, CompositePurchaseOrder compPo) {
+    MonetaryAmount estimatedPrice = calculateEstimatedPrice(poLine.getCost());
+    Transaction transaction = new Transaction();
+    transaction.setTransactionType(Transaction.TransactionType.ENCUMBRANCE);
+    transaction.setFromFundId(distribution.getFundId());
+    transaction.setSource(Transaction.Source.USER);
+    transaction.setCurrency(poLine.getCost().getCurrency());
```

Comment:
The transaction currency must be the system currency , better to set it from the FY currency
> currency - currency code for this transaction - from the system currency.

Reply:
Should we than do conversion? Because for amount calculation  we use `poLine.cost,currency`.